extensions/amp-date-countdown/1.0/test/test-amp-date-countdown.js

```diff
@@ -0,0 +1,330 @@
+    },
+  },
+  (env) => {
+    let win;
+    let element, template;
+    let originalDateNow;
+
+    async function waitRendered() {
+      await whenUpgradedToCustomElement(element);
+      await element.build();
+      await waitFor(() => {
+        // The rendered container inserts a <div> element.
+        const div = element.querySelector('div');
+        return div && div.textContent;
+      }, 'wrapper div rendered');
+      return element.querySelector('div');
+    }
+
+    async function getRenderedData() {
+      const wrapper = await waitRendered();
+      return JSON.parse(wrapper.textContent);
+    }
+
+    beforeEach(() => {
+      win = env.win;
+      element = win.document.createElement('amp-date-countdown');
+      template = win.document.createElement('template');
+      template.setAttribute('type', 'amp-mustache');
+      template.content.textContent = JSON.stringify({
+        years: '{{years}}',
+        months: '{{months}}',
+        days: '{{days}}',
+        hours: '{{hours}}',
+        minutes: '{{minutes}}',
+        seconds: '{{seconds}}',
+        d: '{{d}}',
+        dd: '{{dd}}',
+        h: '{{h}}',
+        hh: '{{hh}}',
+        m: '{{m}}',
+        mm: '{{mm}}',
+        s: '{{s}}',
+        ss: '{{ss}}',
+      });
+      element.appendChild(template);
+      element.setAttribute('layout', 'nodisplay');
+
+      // Mock Date.now()
+      originalDateNow = Date.now;
+      const mockedDateNow = () => Date.parse('2018-01-01T08:00:00Z');
+      Date.now = mockedDateNow;
+    });
+
+    afterEach(() => {
+      // Replace Date.now with its original native function
+      Date.now = originalDateNow;
+    });
+
+    it('renders mustache template into element', async () => {
+      element.setAttribute('end-date', '2018-01-01T08:00:10Z');
+      win.document.body.appendChild(element);
+
+      const data = await getRenderedData();
+
+      expect(data['d']).to.equal('0');
+      expect(data['dd']).to.equal('00');
+      expect(data['h']).to.equal('0');
+      expect(data['hh']).to.equal('00');
+      expect(data['m']).to.equal('0');
+      expect(data['mm']).to.equal('00');
+      expect(data['s']).to.equal('11');
+      expect(data['ss']).to.equal('11');
+
+      expect(data['years']).to.equal('Years');
+      expect(data['months']).to.equal('Months');
+      expect(data['days']).to.equal('Days');
+      expect(data['hours']).to.equal('Hours');
+      expect(data['minutes']).to.equal('Minutes');
+      expect(data['seconds']).to.equal('Seconds');
+    });
+
+    it('renders default template into element', async () => {
+      element.setAttribute('end-date', '2018-01-01T08:00:10Z');
+      element.removeChild(template);
+      win.document.body.appendChild(element);
+
+      const wrapper = await waitRendered();
+      expect(wrapper.textContent).to.equal(
+        'Days 00, Hours 00, Minutes 00, Seconds 11'
+      );
+    });
+
+    it('does not rerender', async () => {
```

Comment:
Please include short "does not rerender b/c ..."

Reply:
Updated test name and included a more detailed comment in test body.